### PR TITLE
[bugfix] create mandatory BASEFS and optional DISTFILES_CACHE

### LINKED ISSUE #7: directories defined in poudriere_config are not created during play
`poudriere_config` has some keys that are path to directory, but they are not automatically created.

### PR DESCRIPTION
**Pull request by trombik**
fixes #7

### CODE REVIEW

**Comment by trombik** [comment]
> no test when BASEFS is not defined or empty

what do you mean by this? creating tests in `tests/serverspec`? the current test scheme assumes that `kitchen test` always succeeds. when `BASEFS` is empty, `kitchen test` will fail, and unit tests will not run. you cannot test `assert` tasks in unit tests. what kind of test do you expect?

**Comment by mitsururike** [comment]
'assert task cannot be tested' should be in our documents (Pitfalls?)

**Comment by trombik** [comment]
no, nor it test the value is `None` or `False`. interestingly, `{% if variable is defined %}` returns true if the variable is `None`. unlike unit tests in programming languages, the cost of tests in this testing scheme, or any other tests that use VMs, is not cheap. cheap containers may help but not much. we should have tested not only x86. welcome to combinatorial explosion.

the question is, is it worth testing it? the reason to choose empty case is that it is the most probable case. assigning `undef` to a variable is rather _difficult_ in jinja2.

**Comment by mitsururike** [comment]
if you can use the role with undefined DISTFILES_CACHE, the test should be done.
as it is defined in defaults/main.yml, it seems you cannot make it undefined.
if you cannot make DISTFILES_CACHE undefined, the conditional  '    - poudriere_config_merged.DISTFILES_CACHE is defined' is not necessary?

**Comment by trombik** [comment]
I understood the comment, "no test when DISTFILES_CACHES is not defined or empty", is, kind of, for completeness, e.g. theoretically, it can be `undef` by an author of the role, could be caused by mistakes or whatever.

do you still think the case should be covered by the unit test? if you do not, that commit would be rolled back. if you do, please push your proposal.

**Comment by mitsururike** [comment]
it seems DISTFILES_CACHE is mandatory and the task should fail if it is not defined or empty.

File: README.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ None
 
 ## `poudriere_config_default`
 
+`BASEFS` is mandatory. See `poudriere.conf.sample` for details.
```
Comment (mitsururike) [inline README.md]: s/is/and `DISTFILES_CACHE` are/

**Pull request merged**